core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -934,7 +935,8 @@ private[spark] object JsonProtocol {
     val executorHost = (json \ "Host").extract[String]
     val totalCores = (json \ "Total Cores").extract[Int]
     val logUrls = mapFromJson(json \ "Log Urls").toMap
-    new ExecutorInfo(executorHost, totalCores, logUrls)
+    val worker = mapFromJson(json \ "Worker").toMap
```

Comment:
This is a compatibility concern, actually: in order to handle replay of old logs, we need to treat this field as optional. See the examples in this file plus the block comment at the top of `JsonProtocol` for instructions. Make sure to add a test case which exercises the backwards-compatibility path, too.